payment/payment_test.go

```diff
@@ -111,5 +118,129 @@ func TestFromPublicKeys(t *testing.T) {
 	if p2sh != "XJkohBHRMT8JUknSqCH7aJP9gAuAe9eNLY" {
 		t.Errorf("TestScriptHash: error when encoding script hash")
 	}
+}
+
+func TestPaymentConfidentialPubKeyHash(t *testing.T) {
+	expected := "VTpzxkqVGbraaCz18fQ2GxLvZkupCi2MPtUdt9ygAEeZ8v9gZPtkD5RUc" +
+		"ap55WZ3aVsbUG6TsQvXc8R3"
+	pk1 := "030000000000000000000000000000000000000000000000000000000000000001"
+	pk1Byte, err := hex.DecodeString(pk1)
+	if err != nil {
+		t.Fatal(err)
+	}
+	pubKey1, err := btcec.ParsePubKey(pk1Byte, btcec.S256())
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	pk2 := "030000000000000000000000000000000000000000000000000000000000000001"
+	pk2Byte, err := hex.DecodeString(pk2)
+	if err != nil {
+		t.Fatal(err)
+	}
+	pubKey2, err := btcec.ParsePubKey(pk2Byte, btcec.S256())
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	payment := payment.FromPublicKey(pubKey1, &network.Liquid, pubKey2)
+	assert.Equal(t, expected, payment.ConfidentialPubKeyHash())
+}
+
+func TestPaymentConfidentialScriptHash(t *testing.T) {
+	expected := "VJLCUu2hpcjPaTGMnANXni8wVYjsCAiTEznE5zgRZZyAWXE2P6rz6Dvph" +
+		"BHSn7iz4w9sLb3mFSHGJbte"
+	scriptHash, err := hex.DecodeString(
+		"9f840a5fc02407ef0ad499c2ec0eb0b942fb0086")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	script := append(append([]byte{txscript.OP_HASH160, 0x14}, scriptHash...),
```

Comment:
Why not using directly the encoded script `a9149f840a5fc02407ef0ad499c2ec0eb0b942fb008687` instead of constructing them from the scriptHash?

Reply:
Done